### PR TITLE
Mac taskcluster improvements [ci full]

### PR DESCRIPTION
**Pull request by bendk**
Move OSX Rust setup and library building to toolchain tasks.  This way we can only build them once, rather than in each task that requires it.
    
- Updated `setup-fetched-rust-toolchain.sh` to work with both the docker worker and mac worker.
- Build the NSS/SQJCipher libraries as a toolchain task that other tasks can depend on.  Right now, we're building these libraries in each task.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by tarikeshaq** [comment]
I imagine this is safe to close @bendk ?

**Comment by bendk** [comment]
This one was blocked by the lack of a L3 mac worker.  I believe we're going to get one very soon, so let's keep it around.

**Comment by bendk** [comment]
Looks like this one is working.  The `fetch-nss-artifact` task is failing because the source link has expired, but that's a separate [issue](https://github.com/mozilla/application-services/issues/5681).

I think the optimization is real.  In this branch, the [nimbus-build-fml-*](https://firefox-ci-tc.services.mozilla.com/tasks/dnRI5NfKSeOFoN0q46-zDA) tasks are taking about 5 minutes, on main [they take about 10](https://firefox-ci-tc.services.mozilla.com/tasks/Nh8dEFdURhGrhpOvx7a9qg).  5 minutes speedup per OSX task seems pretty nice.

File: taskcluster/app_services_taskgraph/transforms/toolchain.py [hunk 1]
```diff
@@ -41,3 +41,10 @@ def resolve_keys(config, tasks):
             routes.append(TOOLCHAIN_OLD_INDEX[task['name']].format(sha=sha))
 
         yield task
+
+@transforms.add
+def fixup_mac_worker(config, tasks):
```
Comment (ahal) [inline taskcluster/app_services_taskgraph/transforms/toolchain.py:46]: nit: technically `mac` isn't the commonality here, it's `generic-worker`

But couldn't we just avoid adding `docker-image` for generic-worker tasks in the first place (like you do in the `setup_worker` transforms below)?

Actually looks like the `toolchain` kind is already using those transforms, so I'm not sure this is even necessary?

File: taskcluster/app_services_taskgraph/transforms/worker.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        try:
+            worker_type = task['worker-type']
+        except KeyError:
+            raise ValueError(f"worker-type not set for {task_name}")
+        if worker_type == 'linux':
+            task['worker-type'] = 'b-linux'
+            worker = task.setdefault('worker', {})
+            worker['docker-image'] = {'in-tree': 'linux'}
+        elif worker_type == 'osx':
+            task['worker-type'] = 'b-osx'
```
Comment (ahal) [inline taskcluster/app_services_taskgraph/transforms/worker.py:22]: I think I would avoid translating the `worker-type` here as it just adds another layer of indirection. If you want to get rid of the `b-` prefix, you can rename the aliases here:
https://github.com/mozilla/application-services/blob/main/taskcluster/ci/config.yml#L17

You could continue setting up the `docker-image` here if you like however.

**Review by ahal — APPROVED** [review]
Looks good!

I have a couple suggestions, though no need for re-review later

File: taskcluster/app_services_taskgraph/transforms/toolchain.py (hunk 1 above)
Comment (bendk) [reply taskcluster/app_services_taskgraph/transforms/toolchain.py]: I'm pretty sure you're right it wasn't necessary anymore after I added the worker transform.  I just deleted that code and am doing one last `[full ci]` check to make sure.

**Pull request merged**